src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -151,15 +151,16 @@
   "Returns a transducer for computing metatdata about the fields in `table`."
   [driver table]
   (map-indexed (fn [i {:keys [database-type], column-name :name, :as col}]
-                 (let [semantic-type (calculated-semantic-type driver column-name database-type)]
+                 (let [base-type (database-type->base-type-or-warn driver database-type)
+                       semantic-type (calculated-semantic-type driver column-name database-type)]
                    (merge
                     (u/select-non-nil-keys col [:name :database-type :field-comment :database-required])
-                    {:base-type         (database-type->base-type-or-warn driver database-type)
+                    {:base-type         base-type
                      :database-position i}
                     (when semantic-type
                       {:semantic-type semantic-type})
                     (when (and
-                           (isa? semantic-type :type/SerializedJSON)
+                           (isa? base-type :type/JSON)
```

Comment:
This doesn't change behaviour because for postgres and mysql, `calculated-semantic-type` returns `:type/SerializedJSON` for the same set of database-types as `database-type->base-type-or-warn` now returns `:type/JSON`. And nothing is derived from `type/SerializedJSON` or `type/JSON`, so `isa?` is essentially doing an equality comparison.

Reply:
This is tested here https://github.com/metabase/metabase/pull/28849/files#diff-3a4e38d096a4324aba5b5c21d7d59324b6becc0792cdff5b9c58dad8ea72bff2R112